packages/hathor-rpc-handler/src/rpcMethods/signWithAddress.ts

```diff
@@ -73,8 +73,8 @@ export async function signWithAddress(
   };
 
   const prompt: SignMessageWithAddressConfirmationPrompt = {
+    ...rpcRequest,
```

Comment:
Hey, I think you should pass the validated params (after zod) on all methods

Because in some cases we are not only validating but also transforming the data (e.g. string -> bigint)